syserol/simulated.py

```diff
@@ -0,0 +1,54 @@
+""" Generate a simulated tensor for testing methods """
+
+from sklearn import impute
+import tensorly as tl
+import seaborn as sns
+import pandas as pd
+import numpy as np
+from syserol.figures.common import getSetup
+from syserol.figures.figure1 import makeFigure
+from syserol.tensor import build_cFactor, continue_R2X, perform_CMTF, curve
+from syserol.COVID import Tensor4D, dayLabels
+from tensorpack.decomposition import Decomposition, entry_drop
+
+
+def generate_simulated(r=6):
+    tensor, _ = Tensor4D()
+    # Generate random CP tensor, in factor form with our COVID tensor shape
+    random_cp = tl.random.random_cp(shape=tensor.shape, rank=r)
+
+    # initialize time factor and continuous factor
+    random_cp.time = dayLabels()
+    # Generate P matrix with realistic values.
+    # Based on curve simulation, a, b, c, and d can/will be in the ranges below
+    P = np.zeros((4, r))
+    for i in range(r):
+        a = np.random.uniform(0.01, 1)
+        b = np.random.uniform(0.01, 1.2)
+        c = np.random.uniform(0.01, 30)
+        d = np.random.uniform(0.01, 100)
+        P[:, i] = a, b, c, d
```

Comment:
Also, here you can see what I've done for the P matrix for now, which is _semi_-manual. I have set these manual parameter ranges that it is currently randomly selecting from, but Cyrillus and I talked yesterday about picking a specific set of these that will generate different looking curves for before&after factorization analysis.